src/shell/notification.js

```diff
@@ -7,14 +7,18 @@ import GLib from 'gi://GLib';
 import GObject from 'gi://GObject';
 import St from 'gi://St';
 
+import {PACKAGE_VERSION} from 'resource:///org/gnome/shell/misc/config.js';
 import * as Main from 'resource:///org/gnome/shell/ui/main.js';
 import * as MessageTray from 'resource:///org/gnome/shell/ui/messageTray.js';
-import * as Calendar from 'resource:///org/gnome/shell/ui/calendar.js';
 import * as NotificationDaemon from 'resource:///org/gnome/shell/ui/notificationDaemon.js';
 
 import {gettext as _} from 'resource:///org/gnome/shell/extensions/extension.js';
 import {getIcon} from './utils.js';
 
+const {NotificationMessage} = Number(PACKAGE_VERSION.split('.')[0]) >= 48
+    ? await import('resource:///org/gnome/shell/ui/messageList.js') // GNOME 48
+    : await import('resource:///org/gnome/shell/ui/calendar.js'); // GNOME 46/47
+
```

Comment:
\*grumble\* I knew notifications were never going to stay part of the `Calendar` code because that simply made not one iota of sense. So, this is a good change, and really I'm annoyed because we went through two releases where all notifications _WERE_ `Calendar.NotificationMessage`s, before that finally got fixed!